src/sql/parts/connection/connectionDialog/connectionDialogWidget.ts

```diff
@@ -244,7 +245,9 @@ export class ConnectionDialogWidget extends Modal {
 
 	private cancel() {
 		this._onCancel.fire();
-		this.close();
+		if (!this._databaseDropdownExpanded) {
```

Comment:
it would be nice if there was a more generic way to determine what ESC should do on the dialog, since this won't scale well if we need other controls to handle ESC as well.  Though given that we don't have another concrete need at this time it's probably fine.